hri_safety_sense/src/VscProcess.cpp

```diff
@@ -36,186 +35,215 @@
 {
-	int retVal = 0;
+  int retVal = 0;
 
-	if(recvMsg.msg.length == sizeof(HeartbeatMsgType)) {
-		ROS_DEBUG("Received Heartbeat from VSC");
+  if(recvMsg.msg.meta.length == sizeof(HeartbeatMsgType)) {
+    RCLCPP_DEBUG(this->get_logger(), "Received Heartbeat from VSC");
 
-		HeartbeatMsgType *msgPtr = (HeartbeatMsgType*)recvMsg.msg.data;
+    HeartbeatMsgType *msgPtr = (HeartbeatMsgType*)recvMsg.msg.meta.data;
 
-		// Publish E-STOP Values
-		std_msgs::UInt32 estopValue;
-		estopValue.data = msgPtr->EStopStatus;
-		estopPub.publish(estopValue);
+    // Publish E-STOP Values
+    std_msgs::msg::UInt32 estopValue;
+    estopValue.data = msgPtr->EStopStatus;
+    estopPub->publish(estopValue);
 
-		if(msgPtr->EStopStatus > 0) {
-			ROS_WARN("Received ESTOP from the vehicle!!! 0x%x",msgPtr->EStopStatus);
-		}
+    if(msgPtr->EStopStatus > 0) {
+      RCLCPP_WARN(this->get_logger(), "Received ESTOP from the vehicle!!! 0x%x", msgPtr->EStopStatus);
+    }
 
-	} else {
-		ROS_WARN("RECEIVED HEARTBEAT WITH INVALID MESSAGE SIZE! Expected: 0x%x, Actual: 0x%x",
-				(unsigned int)sizeof(HeartbeatMsgType), recvMsg.msg.length);
-		retVal = 1;
-	}
+  } else {
+    RCLCPP_WARN(this->get_logger(), "RECEIVED HEARTBEAT WITH INVALID MESSAGE SIZE! Expected: 0x%x, Actual: 0x%x",
+        (unsigned int)sizeof(HeartbeatMsgType), recvMsg.msg.meta.length);
+    retVal = 1;
+  }
 
-	return retVal;
+  return retVal;
 }
 
 void VscProcess::readFromVehicle()
 {
-	VscMsgType recvMsg;
-
-	/* Read all messages */
-	while (vsc_read_next_msg(vscInterface, &recvMsg) > 0) {
-		/* Read next Vsc Message */
-		switch (recvMsg.msg.msgType) {
-		case MSG_VSC_HEARTBEAT:
-			if(handleHeartbeatMsg(recvMsg) == 0) {
-				lastDataRx = ros::Time::now();
-			}
-
-			break;
-		case MSG_VSC_JOYSTICK:
-			if(joystickHandler->handleNewMsg(recvMsg) == 0) {
-				lastDataRx = ros::Time::now();
-			}
-
-			break;
-
-		case MSG_VSC_NMEA_STRING:
-//			handleGpsMsg(&recvMsg);
-
-			break;
-		case MSG_USER_FEEDBACK:
-//			handleFeedbackMsg(&recvMsg);
-
-			break;
-		default:
-			errorCounts.invalidRxMsgCount++;
-			ROS_ERROR("Receive Error.  Invalid MsgType (0x%02X)",recvMsg.msg.msgType);
-			break;
-		}
-	}
-
-	// Log warning when no data is received
-	ros::Duration noDataDuration = ros::Time::now() - lastDataRx;
-	if(noDataDuration > ros::Duration(.25)) {
-		ROS_WARN_THROTTLE(.5, "No Data Received in %i.%09i seconds", noDataDuration.sec, noDataDuration.nsec );
-	}
+  VscMsgType recvMsg;
+
+  /* Read all messages */
+  while (vsc_read_next_msg(vscInterface, &recvMsg) > 0) {
+    /* Read next Vsc Message */
+    switch (recvMsg.msg.meta.msgType) {
+    case MSG_VSC_HEARTBEAT:
+      if(handleHeartbeatMsg(recvMsg) == 0) {
+        lastDataRx = this->now();
+      }
+
+      break;
+    case MSG_VSC_JOYSTICK:
+      if(joystickHandler->handleNewMsg(recvMsg) == 0) {
+        lastDataRx = this->now();
+      }
+
+      break;
+
+    case MSG_VSC_NMEA_STRING:
+//      handleGpsMsg(&recvMsg);
```

Comment:
I know this was here, but maybe get rid of this commented out code and instead print a WARN message?  I guess it depends on whether that is really noisy or not.